### ISSUE TITLE
Add set-env type script for use with git bash that comes with conEmu label:enhancement 

### ISSUE THREAD

**Issue by paul42** [tknerr/bills-kitchen#85]
I really enjoy using the release and the cmd.exe windows shell in con emu works perfectly, but there are certain times I wish I could use git-bash (and the openssl commands, etc.) and have the environment variables set up in the same manner when I launch it - is there a sort of profile that travels with the portablegit? I was wondering what the development effort would be to construct a shell script or if you have any resources that you could point me in for programming a shell script and I could make apull request.

Thanks again!

**Comment by tknerr** [comment]
Hi @paul42, can you try running `W:\tools\portablegit\git-bash.bat` (its not in the PATH yet) inside a `set-env`ed ConEmu shell?

I do that sometimes when I need a more bash like environment and access to all the msysgit additional binaries. Not sure though if that behaves like a fully functional Git Bash (not using it that often), but I assume it would.

If that works for you, we could simply add `W:\tools\portablegit\` to the PATH so its easier accessible.

**Comment by paul42** [comment]
@tknerr works like a charm! thanks for the tip, I'm going to close this issue since it's fixable with a minimum of fuss.

**Comment by paul42** [comment]
hmm, I may have closed it too soon I'm getting a weird error when I attempt to run certain commands - is my path too polluted? I've created a GIST with the command, the error, and the gem environment 
https://gist.github.com/paul42/33fb1cba8fbdd8baa984

**Comment by tknerr** [comment]
@paul42 does `knife openstack flavor list` work in a plain ConEmu cmd.exe shell? Or do you get the same error here?

And can you add the result of running `env` in the gist?

**Comment by paul42** [comment]
Yes, knife openstack flavor list works from the cmd.exe, I've put an 'env' command gist up here:
https://gist.github.com/paul42/55eb0f6cc51376c613bc
I assume you mean from the git-bash, right? I don't know if env works on cmd.exe

**Comment by tknerr** [comment]
@paul42 oh I hate these PATH issues... but let's give it a try...

One thing I'm wondering about is that you got that far. 

With Bills Kitchen 2.3, it looks like this for me in ConEmu:
```
W:\>knife -v
Chef: 11.18.0

W:\>which knife
W:\tools\chefdk\bin\knife
```

...and I get this error with git-bash in ConEmu:
```
tkn@FRAN0508 /w
$ knife -v
bash: /w/tools/chefdk/bin/knife: C:/opscode/chefdk/embedded/bin/ruby.exe: bad interpreter: No such file or directory

tkn@FRAN0508 /w
$ which knife
/w/tools/chefdk/bin/knife
```

So from the `which knife` it looks like the `PATH` is OK. 

The bad interpreter error stems from a ChefDK issue (hardcoded path to "C:/opscode/..."). When I change the first line in `W:\tools\chefdk\bin\knife` to this:
```
#!/w/tools/chefdk/embedded/bin/ruby.exe
...
```

Then `knife -v` actually works in git-bash:
```
tkn@FRAN0508 /w
$ knife -v
Chef: 11.18.0
```

Does a simple `knife -v` work for you actually? (Lets try it with that simple example first since I don't have knife openstack installed)

**Comment by paul42** [comment]
altering the knife file works like charm! knife openstack flavor list now works identically on cmd.exe and git-bash!
https://gist.github.com/paul42/cc12816fda438526eb03

even after altering the knife file, the knife commands and knife openstack flavor list commands work on both git-bash and cmd.exe. this change appears to work perfectly for me, thanks for figuring it out!

**Comment by paul42** [comment]
I was about to create a pull request but it looks like the knife file is created during the  rake process, and I don't know enough ruby/rake to suggest a change for that file. so I leave the hard work up to you :D

**Comment by tknerr** [comment]
Hi @paul42, no worries, I know exactly where to change it in the Rakefile. I will ship with the next release (which I'm probably going to cut once ChefDK 0.5.0 final is out...)

**Comment by tknerr** [comment]
This is fixed via #87 and will ship with the next release. Thanks for the suggestion @paul42!

**Comment by tknerr** [comment]
@paul42 FYI this is released as 3.0-rc1:
https://github.com/tknerr/bills-kitchen/releases/tag/3.0-rc1
